cf-revision-manager.php

```diff
@@ -2,200 +2,450 @@
+				<div>
+				<ul id="cfr_revision_manager_keys">';
+			foreach ( $keys as $key ) {
+				$checked = ( in_array( $key, cfrm_get_selected_keys() ) ? $key : '' );
+				$disabled = '';
+				$id = 'cf_revision_manager_key_' . esc_attr( $key );
+				echo '<li>
+					<input type="checkbox" name="revision_manager_keys[]" id="' . $id . '" value="' . esc_attr( $key ) . '" ' . checked( $key, $checked, false ) . ' ' . disabled( $key, $disabled, false ) . ' />
+					<label for="' . $id . '">' . esc_html( $key ) . '</label>
+				</li>';
 			}
-			return true;
-		}
-	
-		/**
-		 * This is a paranoid check. There will be no object to register the 
-		 * actions and filters if nobody adds any postmeta to be handled
-		 *
-		 * @return bool
-		 */
-		public function have_keys() {
-			return (bool) count($this->postmeta_keys);
+			echo '</ul>
+				</div>
+				<p class="submit">
+				<input type="submit" name="submit_button" class="button-primary" value="' . __( 'Save', 'cfrm' ) . '" />
+				</p>
+				<input type="hidden" name="cf_action" value="cfr_save_keys" class="hidden" style="display: none;" />
+				' . wp_nonce_field( 'cfr_save_keys', '_wpnonce', true, false ) . wp_referer_field( false ) . '
+			</form>';
 		}
-	
-
-		/**
-		* The opposite of WordPress stripslashes_deep, since wp_slash
-		* only works on arrays of strings.
-		**/
-		public function slash_deep($value) {
-			if (is_array($value)) {
-				$value = array_map(array($this, 'slash_deep'), $value);
-			}
-			else if (is_object($value)) {
-				$vars = get_object_vars($value);
-				foreach ($vars as $key => $data) {
-					$value->{$key} = $this->slash_deep($data);
-				}
-			} else if (is_string($value)) {
-				$value = wp_slash($value);
+?>
+</div>
+<?php
+}
+
+/**
+ * Fetch all meta key from postmeta.
+ *
+ * @since  2.0.0
+ *
+ * @return array returns meta key.
+ */
+function cfrm_get_distinct_meta_keys() {
+	global $wpdb;
+	return $wpdb->get_col("
+		SELECT DISTINCT `meta_key`
+		FROM $wpdb->postmeta
+		ORDER BY `meta_key`
+	");
+}
+
+/**
+ * Assign registered key in global variable.
+ *
+ * @since  2.0.0
+ *
+ * @return array returns registered key.
+ */
+function cfrm_set_required_keys() {
+	global $CFR_KEYS_REQUIRED;
+	return $CFR_KEYS_REQUIRED;
+}
+
+/**
+ * Add filter to exclude meta keys.
+ *
+ * @since  2.0.0
+ *
+ * @return array returns meta key.
+ */
+function cfrm_excluded_keys() {
+	return apply_filters(
+		'cf_revision_manager_excluded_keys',
+		array(
+			'_edit_last',
+			'_edit_lock',
+		)
+	);
+}
+
+/**
+ * Return selected meta keys.
+ *
+ * @since  2.0.0
+ *
+ * @return array returns selected meta key.
+ */
+function cfrm_get_selected_keys() {
+	$selected = get_option( 'cf_revision_manager_meta_keys' );
+	if( empty( $selected ) ) {
+		$selected = array();
+	}
+	return $selected;
+}
+
+/**
+ * Register custom meta.
+ *
+ * @since  2.0.0
+ *
+ * @return void
+ */
+function cfrm_register_meta() {
```

Comment:
Why are we registering custom meta?

Why does this function, hooked at init, depend on a global variable?